src/Compiler/CompilerUtils.cpp

```diff
@@ -631,6 +630,13 @@ void addKrnlToAffinePasses(mlir::PassManager &pm) {
 void addKrnlToLLVMPasses(mlir::OpPassManager &pm) {
   pm.addNestedPass<FuncOp>(mlir::createConvertVectorToSCFPass());
   pm.addPass(mlir::createLowerAffinePass());
+
+  // Use MLIR buffer deallocation pass to emit buffer deallocs.
+  // Currently this has to be done *after* lowering the affine dialect because
+  // operations in that dialect do not conform to the requirements explained in
+  // https://mlir.llvm.org/docs/BufferDeallocationInternals.
+  pm.addPass(mlir::createBufferDeallocationPass());
+
```

Comment:
@etiotto I believe Lines 615--620 need to be moved as well. They are for memory bundling and need `dealloc` to analyze buffer lifetime.

Reply:
ok let's try that. I was not sure so thanks for pointing that out @tungld

Reply:
After discussion with @AlexandreEichenberger I have made `--disable-memory-bundling` default to false.

Reply:
> After discussion with @AlexandreEichenberger I have made --disable-memory-bundling default to false.

I guess you mean set to `true` by default. If so, may we change its name from `--disable-memory-bundling` to `--enable-memory-bundling`, which is easy to understand?

Reply:
I can do that, It occurred to me as well. I did not change the name of the option because we have users that use the option, if we are to change the name than I would suggest `--memory-bunding=[on|off]`

Reply:
done